src/Hyperion/SerializerFactories/ArraySerializerFactory.cs

```diff
@@ -68,7 +71,8 @@ private static void ReadValues<T>(int length, Stream stream, DeserializerSession
                     session.TrackSerializedObject(arr);
                 }
 
-                WriteValues((dynamic)arr, stream, elementType, elementSerializer, session);
+                // This janky way of converting array to Array is done to get around the problem of ValueType arrays
+                WriteValues(((IEnumerable)arr).Cast<object>().ToArray(), stream, elementType, elementSerializer, session);
```

Comment:
The only way to convert an array of user defined struct object into an array is to cast it as an `IEnumerable` and then cast all of its elements individually into objects. This is done to convert an array of values into an array of reference objects so that it conforms to the generated IL code.

Reply:
I think this is less janky that casting to a `dynamic` type